src/block/RedstoneWire.php

```diff
@@ -23,12 +23,31 @@
 
 namespace pocketmine\block;
 
+use pocketmine\item\Item;
+use pocketmine\math\Facing;
+use pocketmine\math\Vector3;
 use pocketmine\block\utils\AnalogRedstoneSignalEmitterTrait;
 use pocketmine\block\utils\BlockDataSerializer;
+use pocketmine\world\BlockTransaction;
 
 class RedstoneWire extends Flowable{
 	use AnalogRedstoneSignalEmitterTrait;
 
+	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		$down = $this->getSide(Facing::DOWN);
+		if($down->isSolid()){
```

Comment:
also, check if the block is not a transparent block aka opacity block.

Reply:
https://minecraft.fandom.com/wiki/Opacity
Java blocks won't have any differences right?